### DISCUSSION TITLE
compiling a typescript file produces empty object

### DISCUSSION THREAD

**Discussion by Dmitry-N-Medvedev** [swc-project/swc#9755] [Q&A]
good morning, nice community!

I cannot understand why a compilation of a typescript file gives me a javascript file with an empty object.
Desperately need any ideas/hints/etc. Much appreciated any in advance!

The following `types.ts`
```
//types.ts

import { ByteBuffer } from './byte-buffer.js'
import { Builder } from './builder.js'

export type Offset = number;

export type Table = {
  bb: ByteBuffer
  bb_pos: number
};

export interface IGeneratedObject {
  pack(builder:Builder): Offset
}

export interface IUnpackableObject<T> {
  unpack(): T
}
```

...is translated to the `types.js`:
```
// types.js

export { };
```

the following command has been used to run the compilation:
```
node node_modules/@swc/cli/bin/swc.js ./types.ts --out-dir ./out --sync --config-file .swcrc
```

This is the `.swcrc` file used as a config:

```
// .swcrc

{
  "$schema": "https://swc.rs/schema.json",
  "exclude": [
    "node_modules"
  ],
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "decorators": false,
      "dynamicImport": true
    },
    "transform": {},
    "target": "es2022",
    "loose": false,
    "externalHelpers": false,
    "keepClassNames": true,
    "baseUrl": "./"
  },
  "minify": false,
  "isModule": true,
  "module": {
    "type": "es6",
    "strict": false,
    "strictMode": false,
    "lazy": false,
    "noInterop": false,
    "ignoreDynamic": true,
    "importInterop": "swc",
    "preserveImportMeta": true,
    "resolveFully": true
  }
}
```

**Comment by kwonoj** [comment]
Everything you exported in the module is type or interface, which is expected to be removed after trnaspiling into javascript.

**Reply by Dmitry-N-Medvedev** [reply]
aha, understood! thank you.

then I have a much bigger problem. In the `flatbuffers.js` file there is an import that references the `types.js` file:

```
// flatbuffers.js

export { SIZEOF_SHORT } from "./constants.js";
export { SIZEOF_INT } from "./constants.js";
export { FILE_IDENTIFIER_LENGTH } from "./constants.js";
export { SIZE_PREFIX_LENGTH } from "./constants.js";
export { Table, Offset, IGeneratedObject, IUnpackableObject } from "./types.js";
export { int32, float32, float64, isLittleEndian } from "./utils.js";
export { Encoding } from "./encoding.js";
export { Builder } from "./builder.js";
export { ByteBuffer } from "./byte-buffer.js";
```

Now, when I have an empty object exported from the `types.js` file, the `Table, Offset, etc` aren't going to be found.

I am obviously doing something very naive or even stupid and I have no idea how to even approach this case.

@kwonoj , could you please share you thoughts on this?

**Reply by kdy1** [reply]
You can use type-only import/export specifiers for the imports.

**Reply by Dmitry-N-Medvedev** [reply]
thank you @kdy1 :)

You might have already noticed I am quite new to typescript. Could you please explain your suggestion in a slightly different words or point me to a respective article? Much appreciated!

**Reply by kdy1** [reply]
Ah, I was on mobile as it was about time to sleep. 
```ts
export type { Table, Offset, IGeneratedObject, IUnpackableObject } from "./types.js";
```

or

```ts
export { type Table, type Offset, type IGeneratedObject, type IUnpackableObject } from "./types.js";
```

will work
